### PR TITLE
fix(icons): add missing prop-types dependency

### PR DESCRIPTION
**Pull request by ismay**
The prop-types library is used in the generated react components: https://github.com/dhis2/ui/blob/d7949dad5397c1fe5d1568892c715ded233ec29c/packages/icons/templates/icon-template.js#L16

Also added react and react-dom peer dependencies.

### CODE REVIEW

**Comment by amcgee** [comment]
Nice @ismay thanks for this!  I think this looks good - these are the "patched dependencies" I had to specify when using yarn2 which is more strict, do we still need `ui-constants`?

https://github.com/dhis2/app-platform/pull/491/files#diff-f28e4a8116e6aa62c7ac4cdf57b0040c11bad39d45e24024f0da20ef45972715

**Comment by ismay** [comment]
> Nice @ismay thanks for this! I think this looks good - these are the "patched dependencies" I had to specify when using yarn2 which is more strict, do we still need `ui-constants`?
> 
> https://github.com/dhis2/app-platform/pull/491/files#diff-f28e4a8116e6aa62c7ac4cdf57b0040c11bad39d45e24024f0da20ef45972715

Ah, no ui-constants and styled-jsx shouldn't be necessary anymore. All we're using currently is react and @dhis2/prop-types. I included react-dom as we do that in the other packages as well. I assume that's for compatibility reasons.

**Comment by ismay** [comment]
As an example, this is what one of the generated components looks like:

Screenshot 2021-01-11 at 17 01 01

(the react import comes from the dynamic part of the template, the prop-types lib is hardcoded in our template)

File: yarn.lock [hunk 1]
```diff
@@ -1732,7 +1732,7 @@
     i18next "^10.3"
     moment "^2.24.0"
 
-"@dhis2/prop-types@^1.6.4":
+"@dhis2/prop-types@1.6.4", "@dhis2/prop-types@^1.6.4":
```
Comment (amcgee) [inline yarn.lock:1735]: Side note: where is this "pinned" @dhis2/prop-types 1.6.4 coming from?

**Review by amcgee — APPROVED** [review]
OK cool - I think the overrides were from before the "new" `ui-icons`

File: yarn.lock (hunk 1 above)
Comment (ismay) [reply yarn.lock:1735]: Ah, I probably need to regenerate the lockfile. I'll check.

Comment (ismay) [reply yarn.lock]: Fixed, was a remnant of the install I did in ui-icons where I specified the exact version.

**Pull request merged**